### DISCUSSION TITLE
How to use a texture atlas.

### DISCUSSION THREAD

**Discussion by RylanYancey** [bevyengine/bevy#5718] [Q&A]
Hi, I've read the texture atlas example, but it is confusing and does not help me for my purposes.

Here, I've loaded the texture atlas from my file system:

```rs
use bevy::prelude::*;

pub fn create_cube (
    asset_server: Res<AssetServer>,
) {
    let atlas_handle: Handle<TextureAtlas> = asset_server.load("textures/GrassDirtStoneTree.png");

    // ?? how to use texture atlas ?? //
}
```

What I want to do is split this texture atlas up by sprite size, in this case each sprite is 16x16, and be able to use UVs to index for mesh creation with a PrimitiveTopology. How do I do this?

Also, how do I tell the mesh to use the texture atlas?

TL;DR - How do I extract uvs from a TextureAtlas for mesh creation.

I'm still new to bevy, and trying to understand this practically undocumented engine. I've created a PrimitiveTopology cube before using an RGB value, but not with a texture. 

Thank you very much for your time.

**Comment by bjorn3** [comment]
Duplicate of https://github.com/bevyengine/bevy/discussions/5716, right?

**Reply by rparrett** [reply]
I don't think so. This seems more like confusion about how to texture different faces of a cube.

**Comment by rparrett** [comment]
[`TextureAtlas`](https://docs.rs/bevy/latest/bevy/prelude/struct.TextureAtlas.html) is primarily used for 2d workflows, especially sprite sheet animations and tilemaps.

It is probably possible to use it the way you describe, but it would involve iterating over [`textures`](https://docs.rs/bevy/latest/bevy/prelude/struct.TextureAtlas.html#structfield.textures) and doing the math yourself.

A more typical 3d workflow would involve authoring the UVs in e.g. Blender.

**Reply by RylanYancey** [reply]
ok. I'll try some stuff with Textures and see what happens.
